src/lib.rs

```diff
@@ -3710,3 +3671,56 @@ impl AsmFileExt {
         None
     }
 }
+
+struct PrintThread {
+    handle: Option<JoinHandle<()>>,
+    pipe_writer: Option<os_pipe::PipeWriter>,
+}
```

Comment:
I can make forwarding warnings more efficient by creating and re-use the same printing thread for all `cc::Build::compile_objects` call, but then I cannot ensure all warnings are forwarded.

This might be useful if users call `cc::Build::compile_objects multiple times, but not sure whether it's worth losing some warnings.

Reply:
I think it's fine as-is for now.